### PR TITLE
677 Add hybrid programming example for bundle

### LINKED ISSUE #677: Add bundle example to show hybrid programming
**Is your feature request related to a problem? Please describe.**
As MONAI bundle is very flexible and easy to integrate in hybrid programming (python + JSON/YAML).
Would be nice to add an example to demonstrate that.

### PR DESCRIPTION
**Pull request by Nic-Ma**
Fixes #677 .

### Description
This PR added 2 hybrid programming examples for bundle:
1. Apply customized python component in the train config.
2. Parse config in your own python program and execute training.

### Status
**Ready**

### Checks

- [ ] Notebook runs automatically `./runner [-p <regex_pattern>]`

### CODE REVIEW

File: modules/bundle/README.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# MONAI bundle
+This folder contains the get_started tutorial and concrete training / inference examples for MONAI bundle features.
+
+### [spleen segmentation](./spleen_segmentation)
+A bundle example for volumetric (3D) segmentation of the spleen from CT image.
+### [customize component](./custom_component)
+Example shows the use case that brings customized python component(like: transform, network, metrics, etc.) in the `train` config.
```
Comment (wyli) [inline modules/bundle/README.md:7]:
```suggestion
Example shows the use cases of bringing customized python components, such as transform, network, and metrics, in a configuration-based workflow.
```

File: modules/bundle/custom_component/README.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# Description
+This example mainly shows a typical use case that brings customized python component(like: transform, network, metrics, etc.) in the `train` config.
```
Comment (wyli) [inline modules/bundle/custom_component/README.md:2]:
```suggestion
This example mainly shows a typical use case that brings customized python components (such as transform, network, metrics) in a configuration-based workflow.
```

File: modules/bundle/custom_component/README.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+Please note that this example depends on the `spleen_segmentation` bundle example and executes via overriding the config file of it.
+
+## commands example
+Export the customized python code to `PYTHONPATH`:
```
Comment (wyli) [inline modules/bundle/custom_component/README.md:7]: perhaps comments that "To run the workflow with customized components, `PYTHONPATH` should be revised to include the path to the customized component."

and we should make sure the folder `custom_component/scripts` is a valid python module (it has a `__init__.py` file in the folder)

File: modules/bundle/hybrid_programming/scripts/inference.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+from typing import Sequence, Union
+
+import torch
+from monai.bundle import ConfigParser
+from monai.data import decollate_batch
+from monai.utils.enums import CommonKeys
+
+
+def run(config_file: Union[str, Sequence[str]], ckpt_path: str):
```
Comment (wyli) [inline modules/bundle/hybrid_programming/scripts/inference.py:20]: perhaps include the `logging.conf` if it's useful?

File: modules/bundle/hybrid_programming/scripts/inference.py [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    dataloader = parser.get_parsed_content("dataloader")
+    if len(dataloader) == 0:
+        raise ValueError("no data found in the dataloader, please ensure the input image paths are accessable.")
+    inferer = parser.get_parsed_content("inferer")
+    postprocessing = parser.get_parsed_content("postprocessing")
+
+    model.eval()
+    with torch.no_grad():
+        for d in dataloader:
+            images = d[CommonKeys.IMAGE].to(device)
```
Comment (wyli) [inline modules/bundle/hybrid_programming/scripts/inference.py:40]: do you think it'll be more generic if `CommonKeys.IMAGE` is also defined from the config?

**Review by wyli — COMMENTED** [review]
thanks, I put some minor comments inline

File: modules/bundle/hybrid_programming/scripts/inference.py (hunk 4 above)
Comment (Nic-Ma) [reply modules/bundle/hybrid_programming/scripts/inference.py:20]: Hi @wyli ,

`logging.conf` is not useful in this example.
The engines of spleen segmentation bundle can work with `logging.conf`, this example doesn't use engines.

Thanks.

**Review by wyli — APPROVED** [review]

File: modules/bundle/README.md (hunk 1 above)
Comment (Nic-Ma) [reply modules/bundle/README.md:7]: updated in the latest commit.

Thanks.

File: modules/bundle/custom_component/README.md (hunk 2 above)
Comment (Nic-Ma) [reply modules/bundle/custom_component/README.md:2]: updated in the latest commit.

Thanks.

File: modules/bundle/custom_component/README.md (hunk 3 above)
Comment (Nic-Ma) [reply modules/bundle/custom_component/README.md:7]: updated in the latest commit.

Thanks.

File: modules/bundle/hybrid_programming/scripts/inference.py (hunk 5 above)
Comment (Nic-Ma) [reply modules/bundle/hybrid_programming/scripts/inference.py:40]: Sure, updated in the latest commit.

Thanks.

**Pull request merged**